### PR TITLE
Extractor: replace `sandbox-classic`-based fixtures with `sandbox`-based ones

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
All tests that are supposed to check whether the _extracted_ values match what is supposed to be on the ledger fail, like the following:

```
Contracts
- should be extracted *** FAILED *** (64 milliseconds)
  List() had length 0 instead of expected length 3 (BasicPrimitiveTypesSpec.scala:35)
  ...
- should contain the correct JSON data *** FAILED *** (37 milliseconds)
  List() did not contain the same elements as List({
    "reference" : "Simple values",
    "int_field" : 5,
    "decimal_field" : "5.5",
    "numeric0_field" : "42.0",
    "numeric37_field" : "0.25",
    "text_field" : "Hey",
    "bool_field" : true,
    "party_field" : "Bob",
    "date_field" : "2020-02-22",
    "time_field" : "2020-02-22T12:13:14Z"
  }, {
    "reference" : "Positive extremes",
    "int_field" : 9223372036854775807,
    "decimal_field" : "9999999999999999999999999999.9999999999",
    "numeric0_field" : "99999999999999999999999999999999999999.0",
    "numeric37_field" : "9.9999999999999999999999999999999999999",
    "text_field" : "Hey",
    "bool_field" : true,
    "party_field" : "Bob",
    "date_field" : "9999-12-31",
    "time_field" : "9999-12-31T23:59:59Z"
  }, {
    "reference" : "Negative extremes",
    "int_field" : -9223372036854775808,
    "decimal_field" : "-9999999999999999999999999999.9999999999",
    "numeric0_field" : "-99999999999999999999999999999999999999.0",
    "numeric37_field" : "-9.9999999999999999999999999999999999999",
    "text_field" : "Hey",
    "bool_field" : true,
    "party_field" : "Bob",
    "date_field" : "0001-01-01",
    "time_field" : "0001-01-01T00:00:00Z"
  }) (BasicPrimitiveTypesSpec.scala:88)
```

I'm a bit in the dark as to what the cause of this could be.

File: daml-script/test/BUILD.bazel [hunk 1]
```diff
@@ -121,6 +121,7 @@ da_scala_library(
         "@maven//:org_scalatest_scalatest_wordspec",
         "@maven//:org_scalaz_scalaz_core",
     ],
+    visibility = ["//visibility:public"],
```
Comment (stefanobaghino-da) [inline daml-script/test/BUILD.bazel:124]: Re-using the helpers in here to test extractor using scripts to initialize the state.

File: extractor/BUILD.bazel [hunk 2]
```diff
@@ -20,21 +20,44 @@ load(
...
+dependencies:
+  - daml-stdlib
+  - daml-prim
+  - daml-script.dar
+EOF
+      $(location //compiler/damlc) build --project-root=$$TMP_DIR --ghc-option=-Werror -o $$PWD/$(location test.dar)
+      rm -rf $$TMP_DIR
+    """.format(sdk = sdk_version),
+    tools = ["//compiler/damlc"],
+)
```
Comment (stefanobaghino-da) [inline extractor/BUILD.bazel:60]: Looks like there is no easy way of building a DAR with script as a dependency, I copied this and adapted it from the Daml Script tests.

File: extractor/src/test/lib/scala/com/digitalasset/extractor/services/ExtractorFixture.scala [hunk 3]
```diff
@@ -3,39 +3,50 @@
...
+  override protected val timeMode: ScriptTimeMode = ScriptTimeMode.Static
+
+  protected def baseConfig: ExtractorConfig = ExtractorConfig(
     "127.0.0.1",
     ledgerPort = Port(666), // doesn't matter, will/must be overridden in the test cases
     ledgerInboundMessageSizeMax = 50 * 1024 * 1024,
     LedgerOffset(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_BEGIN)),
     SnapshotEndSetting.Head,
-    OneAnd(Party assertFromString "Bob", List.empty),
+    OneAnd.oneAndNelIso.to(parties.map(Party.assertFromString)),
```
Comment (stefanobaghino-da) [inline extractor/src/test/lib/scala/com/digitalasset/extractor/services/ExtractorFixture.scala:49]: I'm sure there is a better way to do this, listening for suggestions.

File: extractor/src/test/resources/damls/EnumMod.daml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- Port of daml-lf/encoder/src/test/lf/Enum_all_.lf
```
Comment (stefanobaghino-da) [inline extractor/src/test/resources/damls/EnumMod.daml:4]: The tests were relying on these Daml-LF tests that use scenario. I don't believe there's an easy way to use script from there, so I ported it. Same for another one.

File: extractor/src/test/resources/damls/EnumMod.daml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+template Box
+  with
+    x: Color
+    party: Party
+  where
+    signatory party
+
+createContracts =
+  script do
+    party <- allocatePartyWithHint "EnumMod" (PartyIdHint "EnumMod")
```
Comment (stefanobaghino-da) [inline extractor/src/test/resources/damls/EnumMod.daml:21]: I relied on hints, which is sub-optimal but should be good enough for the time being. Ideally extractor will be removed before we need to move to Canton.

File: extractor/src/test/suite-flaky/com/daml/extractor/VeryLargeArchiveSpec.scala [hunk 6]
```diff
@@ -50,11 +50,11 @@ class VeryLargeArchiveSpec
   // future editors of this test should not feel obliged to synthesize a failure
   // if the system design has really changed so failures of this nature cannot
   // happen.
-  val failMB = 1
-  val successMB = 10
+  val failBytes = 1024 // 1 KB
```
Comment (stefanobaghino-da) [inline extractor/src/test/suite-flaky/com/daml/extractor/VeryLargeArchiveSpec.scala:53]: I did this on purpose. For some reason the tests kept failing (i.e. despite going above the threshold this succeeded). It's unclear to me what the actual problem is but I wouldn't be surprised if these tests are suddenly no longer flaky.

File: extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala [hunk 7]
```diff
@@ -49,14 +52,13 @@ class TransactionMultiTableSpec
         case TransactionResult(
               transaction_id,
               seq,
-              workflow_id,
+              _, // Daml Script leaves the workflow identifier empty
```
Comment (stefanobaghino-da) [inline extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala:55]: Daml Script doesn't use the workflow identifier like scenarios apparently, so I ignored it in all tests.

File: extractor/BUILD.bazel (hunk 2 above)
Comment (cocreature) [reply extractor/BUILD.bazel:60]: totally fine for now, I’ll look into cleaning this up later

File: extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala (hunk 7 above)
Comment (cocreature) [reply extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala:55]: at some point I’ll understand what the workflow id is but that time is not today

**Review by cocreature — APPROVED** [review]
Thanks!

File: extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala (hunk 7 above)
Comment (stefanobaghino-da) [reply extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala:55]: I think that the idea behind it is to correlate transactions within a single logical workflow, e.g. if you have some form of trade that needs to be separated in multiple legs due to some business requirement.

**Review by akshayshirahatti-da — APPROVED** [review]

**Pull request merged**